### DISCUSSION TITLE
FVM-driven gas model changes

### DISCUSSION THREAD

**Discussion by raulk** [filecoin-project/FIPs#316] [Enhancements - Technical]
There are two FIPs that introduce gas model changes within the context of the Filecoin Virtual Machine.

* [FIP-0032: Gas model adjustment for non-programmable FVM](https://github.com/filecoin-project/FIPs/pull/315).
    * Introduces execution gas, rearchitects IPLD state management fees, adjusts the actor call fee.
    * State: soon ready to be accepted, once the benchmarks that @stebalien is running are in.
    * Expected activation: nv16 (Skyr upgrade, introduces non-programmable FVM).
* [FIP-nnnn: Gas model adjustment for user-programmability](https://github.com/filecoin-project/FIPs/pull/317).
    * Introduces the actor installation fee, introduces the actor loading fee, introduces memory expansion gas, prices unpriced syscalls, and more.
    * State: WIP, open draft.
    * Expected activation: nv18 (presumably the upgrade that enables user-programmability).

Please use this thread to discuss general aspects on approach, motivation, risks, etc. of these FIPs.

**Comment by anorth** [comment]
Noting here an open discussion from https://github.com/filecoin-project/FIPs/pull/315#discussion_r822256772 that the relative costs of various methods, as well as the total gas consumption for a given level of activity, will both change. We want to understand the product considerations of those (after we know the numbers).

**Comment by anorth** [comment]
Some questions about the second proposal.

- Is `debug_enabled` a thing for real networks? What does it do? If it does anything, it's new and should be described in a FIP somewhere. We'll probably have another one introducing new syscalls anyway, and we could define its gas there.
- The proposal ignores the `resolve_address` syscall. This is another functional change to hash out in a FIP. Some discussion in (https://github.com/filecoin-project/ref-fvm/issues/368)

**Reply by kaitlin-beegle** [reply]
Hey, @anorth! This (and the question above, re: product changes) were opened quite awhile ago.  Are these still open questions, or do you consider them resolved?

**Reply by anorth** [reply]
Yes. But a good note to us FIP editors to check the discussion-to post for each FIP before moving to last call.

**Comment by tmellan** [comment]
CryptoEcon Lab looked into FIP-0032. Since the relative gas usage of sector and batch commits will change, we thought there was scope for this to affect the rational batching unit economics, and in turn the effectiveness of the batch balancer. Based on the evidence we have from the benchmark multipliers for post-FIP-0032 gas usages, the batch balancer is likely to continue to be effective. This is contingent on the benchmark being representative of future operation, so it's good to monitor gas usage post implementation. A more detailed summary with a link to a Jupyter notebook is available on our [Notion page](https://www.notion.so/pl-strflt/FIP-0032-assessment-summary-3b05560a80004578aa2d04b082ffb908).

**Reply by tmellan** [reply]
Patrick from Factor8 Solutions flagged that the Notion page is not viewable without signing in. Sharing a [hackmd copy](https://hackmd.io/@U76Tn8ZiTh-f4b7FjqHlmA/HJ_Xl2Zs5) for easier access.
